AutomatedTesting/Gem/PythonTests/assetpipeline/scene_tests_fixtures/scene_test_debug_compare.py

```diff
@@ -0,0 +1,241 @@
+    # removed due to floating point value drift.
+    initial_diff_actual_len = len(diff_actual)
+    initial_diff_expected_len = len(diff_expected)
+
+    # Decide the number of digits past the decimal point to preserve.
+    DECIMAL_DIGITS_TO_PRESERVE = 3
+
+    # This function requires the two lists to be equal length.
+    assert initial_diff_actual_len == initial_diff_expected_len, "Scene mismatch - different line counts"
+
+    # Store the trimmed values and compare the un-trimmed values separately, emitting warnings.
+    # Trim decimals from the lists to be compared, if any where found, re-compare and generate new lists.
+    floating_point_regex = re.compile(
+        f"(.*?-?[0-9]+\\.[0-9]{{{DECIMAL_DIGITS_TO_PRESERVE},{DECIMAL_DIGITS_TO_PRESERVE}}})[0-9]+(.*)")
+    for index, diff_actual_line in enumerate(diff_actual):
+        # Loop, because there may be multiple floats on the same line.
+        while True:
+            match_result = floating_point_regex.match(diff_actual[index])
+            if not match_result:
+                break
+            diff_actual[index] = f"{match_result.group(1)}{match_result.group(2)}"
+        # diff_actual and diff_expected have the same line count, so they can both be checked here
+        while True:
+            match_result = floating_point_regex.match(diff_expected[index])
+            if not match_result:
+                break
+            diff_expected[index] = f"{match_result.group(1)}{match_result.group(2)}"
+
+    # Re-run the diff now that floating point values have been truncated.
+    diff_actual, diff_expected = utils.get_differences_between_lists(diff_actual, diff_expected)
+
+    # If both lists are now empty, then the only differences between the two scene files were floating point drift.
+    if (diff_actual == None and diff_expected == None) or (len(diff_actual) == 0 and len(diff_actual) == 0):
+        warnings.warn(f"Floating point drift detected between {expected_file_path} and {actual_file_path}.")
+        return diff_actual, diff_expected
+
+    # Something has gone wrong if the lists are now somehow different lengths after the comparison.
+    assert len(diff_actual) == len(diff_expected), "Scene mismatch - different line counts after truncation"
+
+    # If some entries were removed from both lists but not all, then there was some floating point drift causing
+    # differences to appear between the scene files. Provide a warning on that so it can be tracked, then
+    # continue on to the next set of list comparisons.
+    if len(diff_actual) != initial_diff_actual_len or len(diff_expected) != initial_diff_expected_len:
+        warnings.warn(f"Floating point drift detected between {expected_file_path} and {actual_file_path}.")
+
+    return diff_actual, diff_expected
+
+
+def scan_scene_debug_xml_file_for_issues(diff_actual: List[str], diff_expected: List[str],
```

Comment:
Nit: Feel like this function can be simplified a little bit to remove some redundant code: The logic for processing diff_actual and diff_expected are the same and you can probably just call the same function on these two different lists.

Reply:
Added this to the refactoring work for phase 2.